File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    const shortest = new Map();
    
    for (let key of keymap) {
        for (let i = 0; i < key.length; i++) {
            const char = key[i];
            
            if (shortest.has(char)) {
                const short = Math.min(shortest.get(char), i + 1);
                shortest.set(char, short);
            } else {
                shortest.set(char, i + 1);
            }
        }
    }
    console.log(shortest);
    
    const result = [];
    
    for (let target of targets) {
        let sum = 0;
        
        for (let t of target) {
            if (!shortest.has(t)) {
                sum = -1;
                break;
            }
            
            sum += shortest.get(t);
        }
        
        result.push(sum);
    }
    
    return result;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    shortcut = {}
    
    for km in keymap:
        for i in range(len(km)):
            char = km[i]
            
            if char not in shortcut:
                shortcut[char] = i+1
            else:
                shortcut[char] = min(shortcut[char], i+1)
                
    result = []
    
    for target in targets:
        count = 0
        
        for t in target:
            if t not in shortcut:
                count = -1
                break
            else:
                count += shortcut[t]
            
        result.append(count)
        
    return result